src/typescript/13-utility-types/utility-types.md

```diff
@@ -1,117 +1,207 @@
 @page learn-typescript/utility-types Utility Types
 @parent learn-typescript 13
+@outline 3
 
-@description Learn About the Utility Types provided by TypeScript
+@description Learn about the utility types provided by TypeScript
 
 @body
 
 ## Overview
 
-Utility Types are built-in helper Types meant for common Type transformations.
+In this section, you will:
 
-In this section you will see a small description for the most used Utility Types.
+- Understand the purpose of utility types
+- Make properties optional with `Partial<Type>`
+- Make properties required with `Required<Type>`
+- Set properties as immutable with `Readonly<Type>`
+- Map keys to a type with `Record<Keys, Type>`
+- Include specific properties with `Pick<Type, Keys>`
+- Exclude specific properties with `Omit<Type, Keys>`
+- Exclude values from a type with `Exclude<Type, ExcludedUnion>`
+- Select included subtypes with `Extract<Type, Union>`
+- Remove nullability with `NonNullable<Type>`
+- Get function return type with `ReturnType<Type>`
 
-We are going to repeat the interface Dinosaur below throughout the examples.
+## Objective 1: Utility Types
+
+Utility types are generic helpers that have been created to help assist with common type transformations. We could accomplish implementing all of our types without utility types, however, these are provided by TypeScript to help with the common use cases. Through that, utility types allow us to quickly and easily build different types to enhance our type safety. This will make the code more comprehensible and concise.
+
+## Objective 2: Partial&lt;Type&gt;
```

Comment:
Let’s match the fall training: https://docs.google.com/presentation/d/1pTOIZhbEscGv-zyQEJ3U0nwMvqZleNF5/edit#slide=id.g27e5da00130_0_31

> Objective 1: Property existence modifiers